tachyon/zk/plonk/halo2/prover.h

```diff
@@ -147,18 +146,21 @@ class Prover : public ProverBase<PCS> {
     std::vector<MultiPhaseRefTable<Evals>> column_tables =
         argument_data->ExportColumnTables(proving_key.fixed_columns());
 
-    lookup::halo2::Prover<Poly, Evals>::BatchCompressPairs(
-        lookup_provers, domain, cs.lookups(), theta, column_tables);
-    lookup::halo2::Prover<Poly, Evals>::BatchPermutePairs(lookup_provers, this);
-
-    if constexpr (PCS::kSupportsBatchMode) {
-      this->pcs_.SetBatchMode(
-          lookup::halo2::Prover<Poly, Evals>::GetNumPermutedPairsCommitments(
-              lookup_provers));
-    }
     size_t commit_idx = 0;
-    lookup::halo2::Prover<Poly, Evals>::BatchCommitPermutedPairs(
-        lookup_provers, this, commit_idx);
+    if constexpr (LS::type == lookup::Type::kHalo2) {
+      LookupProver::BatchCompressPairs(lookup_provers, domain, cs.lookups(),
+                                       theta, column_tables);
+      LookupProver::BatchPermutePairs(lookup_provers, this);
+
+      if constexpr (PCS::kSupportsBatchMode) {
+        this->pcs_.SetBatchMode(
+            LookupProver::GetNumPermutedPairsCommitments(lookup_provers));
+      }
+      LookupProver::BatchCommitPermutedPairs(lookup_provers, this, commit_idx);
+    } else {
+      NOTREACHED();
```

Comment:
```suggestion
      static_assert(false);
```

Reply:
`static_assert(false)` invariably fails in compile time regardless of the condition.